### DISCUSSION TITLE
Convert from OST to PST?

### DISCUSSION THREAD

**Discussion by petervk** [iluvadev/XstReader#5] [Ideas]
Hi! This program seems to be the only thing I could find that could open an OST file I have, and it seems like I can export all of the emails into HTML format, but what I really need is to convert the OST file (or at least the contents of a number of subfolders) into a PST file so I can get these emails back into Outlook and on to exchange. Is there any way to do that?

**Comment by iluvadev** [comment]
Hi! 
XstReader only can read (and export) ost and pst files, is not in mind write at these formats: there are a lot of considerations, and is not easy, but I'll do some research to try to find some solution or alternative.
For now I think that this program will not be able to meet the needs. But, let's analyze it ;-)

Thanks for this comment, I think it is a fairly common need, but also quite difficult.

**Reply by petervk** [reply]
Yeah, I figured it was complicated as there doesn't even seem much options for even reading them. An interim solution could be to export a csv file that can be imported into Outlook, or exported into Outlook readable *.msg files instead of text or htmp.

**Reply by iluvadev** [reply]
I explored the option to make an ost to pst conversor and honestly, I don't see it clearly: .ost files are a "simplification" of .pst files, for reading we can ignore some structures that are in .pst but not in .ost, for reading purposes they are similar.
If we want to create a .pst file, we need generate those structures. I will explore in more detail, but I find it quite expensive.

The other option is export individual messages in an Outlook readable format. I think is the more realistic option:

- Export messages to *.msg files.  This format is not supported by XstReader, and it seems that .msg files are made up of internal structures similar to those of .pst files. 
- Export messages to *.csv files. Now XstReader can export in CSV all the properties of a single message and all properties of the messages inside a folder. But I think that this csv is not correct to be imported in Outlook.

## Summary
Exporting to .csv might be the option with the lowest development cost.
When I have some time I try it.

Thanks!

**Reply by ShivamRathore99** [reply]
Many users want to open OST files but are unable to do so because they first need to export the OST file into PST to view it. Therefore, we are providing you with the best **OST to PST Converter** that allows you to convert OST files to PST, including emails, contacts, calendars, tasks, notes, journals, etc. It supports the conversion of OST files into PST, PDF, MBOX, CSV, MSG, EML, MHTML, DOC, DOCX, DOCM, etc.  It has advanced features to discard or eliminate duplicate emails.

**Reply by Julypatel** [reply]
**MSOutlookHelp OST to Outlook PST Converter** is an easy-to-use tool that converts OST files to PST file format. It can also convert OST files into multiple formats such as PST, MBOX, EML, EMLX, CSV, DOC, and many more. The tool has multiple features like  Export OST Files to CSV, vCard (VCF) & ICS Formats in which the tool provides an option to export OST file contacts into vCard or CSV format and Calendars into ICS format. With the help of this software, you can search out the attachments of OST email items and save them into a separate folder according to requirements. Moreover, the tool is compatible with both the Mac and Windows operating systems.

**Comment by iluvadev** [comment]
Hi, I just published a new Release, with the ability to export any message in a .msg format. 
With this, you can import messages in Outlook.

**Reply by MKotori** [reply]
Dear iluvadev, thank you & Dijji for the great work. This program is useful.
From OST to PST, the directory index will be retained, while from OST to MSG, it will lose the directory index.
And from OST to PST, from one file to one file; from OST to MSG, from one file to many files.
It should be a very challenging and complicated work for the new function.

By the way, do you have any plan to build multilingual version? I can provide Chinese translation.

**Comment by kenthomson001** [comment]
Thank you for sharing the information.

**Comment by kenthomson001** [comment]
If your Exchange account is still active, you can convert the OST to PST directly through Outlook by going to File → Open & Export → Import/Export → Export to a File → Outlook Data File (.pst).  This is the simplest and safest method with no third-party tools needed.

**Comment by scottjohn487-jpg** [comment]
If you're working with large or older PST files, compatibility and preserving the original mailbox structure can be challenging. It's always a good idea to verify that folders, attachments, and email metadata are retained correctly after opening or converting the file, especially if the data is intended for long-term access or migration
